collect_app/src/main/java/org/odk/collect/android/activities/FirstLaunchActivity.kt

```diff
@@ -76,34 +88,51 @@ class FirstLaunchActivity : LocalizedActivity() {
                     }
 
                 setOnClickListener {
-                    Analytics.log(AnalyticsEvents.TRY_DEMO)
-
-                    val isLoading = MutableNonNullLiveData(true)
                     MaterialProgressDialogFragment.showOn(
                         this@FirstLaunchActivity,
-                        isLoading,
+                        viewModel.isLoading,
                         supportFragmentManager
                     ) {
                         MaterialProgressDialogFragment().also { dialog ->
                             dialog.message = getString(org.odk.collect.strings.R.string.loading)
                         }
                     }
 
-                    scheduler.immediate(
-                        background = {
-                            projectsRepository.save(Project.DEMO_PROJECT)
-                            projectsDataService.setCurrentProject(Project.DEMO_PROJECT_ID)
-                        },
-                        foreground = {
-                            isLoading.value = false
+                    viewModel.isLoading.observe(this@FirstLaunchActivity) { isLoading ->
```

Comment:
This won't work for configuration changes as the observation won't get set up on the recreated Activity (it's only made after clicking "Try demo"). Both the observation for the progress dialog (`showOn`) and this should be in `onCreate`. 

Just a thought on the side: It's often the case with these reactive values that the behaviour is declared "up front" rather than located nearer the events they're related to - that's nice because you could separation of concerns there.

Reply:
Ahh of course... ok now it should be ok.